ct-app/core/components/parameters.py

```
@@ -1,101 +1,298 @@
-        try:
-            value = float(value)
-        except ValueError:
-            pass
+class ExplicitParams:
+    keys: dict[str, type] = {}
 
-        try:
-            integer = int(value)
-            if integer == value:
-                value = integer
+    def __init__(self, data: dict = {}):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)